src/filepath.cpp

```diff
@@ -71,11 +71,10 @@ bool Filepath::starts_with(const std::string& str) const
 nonstd::optional<Filepath> Filepath::file_name() const
 {
 	auto str = filepath::bridged::file_name(*rs_object);
-	auto res = std::string(str.begin(), str.end());
-	if (res.empty()) {
+	if (filepath::bridged::is_empty(*rs_object)) {
 		return nonstd::nullopt;
 	} else {
-		return res;
+		return str;
```

Comment:
This asks the compiler to convert `Box<PathBuf>` into `Filepath` and then convert *that* into `nonstd::optional<Filepath>`. I think you can help older compilers to chew through this with `return {str}` or maybe `return nonstd::optional<Filepath>(str)`.

Reply:
It's not work.

`{str}` leads to
```bash
src/filepath.cpp: In member function ‘nonstd::optional_lite::optional<newsboat::Filepath> newsboat::Filepath::file_name() const’:
src/filepath.cpp:77:14: error: could not convert ‘{str}’ from ‘<brace-enclosed initializer list>’ to ‘nonstd::optional_lite::optional<newsboat::Filepath>’
   77 |   return {str};
      |              ^
      |              |
      |              <brace-enclosed initializer list>
```

`nonstd::optional<Filepath>(str)` can't be constructed

```bash
src/filepath.cpp: In member function ‘nonstd::optional_lite::optional<newsboat::Filepath> newsboat::Filepath::file_name() const’:
src/filepath.cpp:77:40: error: no matching function for call to ‘nonstd::optional_lite::optional<newsboat::Filepath>::optional(rust::cxxbridge1::Box<newsboat::filepath::bridged::PathBuf>&)’
   77 |   return nonstd::optional<Filepath>(str);
      |                                        ^
```

Let's revert commits.